backend/backend.go

```diff
@@ -1155,7 +1181,10 @@ func (backend *Backend) Close() error {
 	errors := []string{}
 
 	backend.ratesUpdater.Stop()
+
+	unlock := backend.accountsLock.Lock()
```

Comment:
I would just do it at the top with defer'ed unlock. After all, it says: "After this, no other method should be called."

Might be irrelevant but I noticed this manipulates `backend.coins`. I think it should also locked for the whole duration and possibly `delete(backend.coins, coin)`?

Reply:
done (not the coins part, maybe in another PR)